Merge example page into welcome screen

jprochazk opened the pull request:
### What

- Part of https://github.com/rerun-io/rerun/issues/4961
- Closes https://github.com/rerun-io/rerun/issues/3244

Changes:
- Updated design of thumbnails on example page to match https://rerun.io/examples
- Updated some spacing on welcome screen to better match design from #3244
- Remove loading spinners on example page thumbnails
- Remove tabs on welcome screen + remove the examples welcome screen card + the whole mechanism of switching between the welcome screen and example page
- Move example page contents below welcome screen
- If example page is not visible, indicate their presence with a floating `See examples` button, which may be clicked to scroll the examples page into view

I have not done anything to change the loading behavior, or added any extra effects as the example page loads in. There are also a few things in #4961 not done in this PR, such as updating the copy or merging the quick start examples.

![image](https://github.com/rerun-io/rerun/assets/1665677/173cbfd6-4f5d-43d8-8b91-e01aa0519376)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5329/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5329/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5329/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5329)
- [Docs preview](https://rerun.io/preview/b23e7aa1a53bc82682b6327c95b56fd600c09a99/docs) <!--DOCS-PREVIEW-->
- [Examples preview](https://rerun.io/preview/b23e7aa1a53bc82682b6327c95b56fd600c09a99/examples) <!--EXAMPLES-PREVIEW-->
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

jprochazk commented:
First pass at updating the example page styles: 
![image](https://github.com/rerun-io/rerun/assets/1665677/7bb2827a-a90d-4623-8ec4-ace8b9af5802)

It's not exactly the same as https://rerun.io/examples, because we want to keep the size of each example, and I couldn't think of a better place to put it.

abey79 commented on crates/re_ui/data/icons/arrow_down.png:
We should make it 28x28px (without necessarily making the arrow itself bigger). It would help for consistency with all other icons, should we end up using it for something else.

abey79 commented on crates/re_viewer/src/ui/welcome_screen/example_page.rs:
```diff
@@ -379,72 +369,109 @@ impl ExamplePage {
+        if !examples_visible {
+            let screen_rect = ui.ctx().screen_rect();
+            let indicator_rect = examples_page_rect
+                .with_min_y(screen_rect.bottom() - 125.0)
```
I know for a fact that it isn't always convenient to chase every last magic number out, but this one is a little bit scary. Is this going to be robust enough to other changes?

I'm not sure it is relevant here, but I sometime "record" the current screen position after some UI elements by storing `ui.cursor().min.y` in a variable which can then be used later on. Maybe that pattern applies here?

abey79 commented on crates/re_viewer/src/ui/welcome_screen/example_page.rs:
```diff
@@ -23,7 +23,7 @@ struct ExampleDesc {
     title: String,
 
-    description: String,
+    // description: String,
```
nit: remove dead code

abey79 commented on crates/re_viewer/src/ui/welcome_screen/example_page.rs:
Maybe rename the file to `example_section.rs` for consistency with the reorganisation.

abey79 reviewed (approved):
This is such a great improvement!

Code generally lgtm, with a few minor comments.

jprochazk commented:
- Changed the `Page` in `ExamplePage` and `WelcomePage` to `Section`, and updated all references
- Did some minor refactoring of the panel buttons in the welcome section, there was some code leftover which was previously used for switching to the example page.
- Updated the icon to 28x28 px
